onnx/defs/data_propagators.h

```diff
@@ -49,4 +53,29 @@ inline void PropagateShapeDataFromInputToOutput(DataPropagationContext& ctx, int
   }
 }
 
+inline void GatherOp13DataPropagator(DataPropagationContext& ctx) {
```

Comment:
nit: seems like this PR is doing two different things? Just wondering why it also has updates to specific data. prop. logic?